### DISCUSSION TITLE
Issues with Firebase Auth Emulator in Flutter API Test

### DISCUSSION THREAD

**Discussion by lukas1001** [firebase/flutterfire#12951] [Q&A]
Hi everyone,

I'm trying to write an API test for Firebase authentication using the Firebase Auth emulator in my Flutter project. Here is what I've done so far:

### SETUP
**Firebase Emulator:**
I have set up the Firebase Auth emulator and it's running. I can access the emulator in my browser at localhost:4000, and I see the active authentication emulator on port 4001.

**Flutter Test Code:**
I wrote a test that initializes Firebase with the emulator settings and then attempts to create a new user.

### Code
```
//file: test\firebase_test.dart

import 'package:firebase_auth/firebase_auth.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:zeitmanagement/firebase_options.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();

  setUpAll(() async {
    await Firebase.initializeApp(
      options: DefaultFirebaseOptions.currentPlatform,
    );
    await FirebaseAuth.instance.useAuthEmulator('localhost', 4001);
  });

  test('Able to create User', () async {
    await FirebaseAuth.instance.createUserWithEmailAndPassword(
      email: 'example@email.com',
      password: 'any123!password',
    );
    final user = FirebaseAuth.instance.currentUser;
    expect(user, isNotNull);
    expect(user?.email, 'example@email.com');
  });
}
```

### Error
When I run the test, I get the following error at the setUpAll() function:
"PlatformException(channel-error, Unable to establish connection on channel., null, null)
package:firebase_core_platform_interface/src/pigeon/messages.pigeon.dart 210:7  FirebaseCoreHostApi.initializeCore"

### Environment
Flutter Version: 3.22.0
Firebase Packages Version: firebase_core: ^2.30.1, firebase_auth: ^4.19.3
OS: Windows 10

Does anyone have insights on what might be causing this issue or suggestions on how to properly configure the Firebase Auth emulator for testing in Flutter?

**Comment by tony8864** [comment]
Hello @lukas1001. I am having exactly the same issue. Were you able to solve it?

**Comment by lukas1001** [comment]
Hi @tony8864
It's a while back so I had to look at my old files again. As I understood it from [this ](https://firebase.flutter.dev/docs/testing/testing/#:~:text=Unit%20tests%20using%20fakes,library%20and%20simulates%20its%20behavior.)article, firebase can't be tested in unit tests, since it needs a real device. So I did all my tests using integration tests - [here ](https://github.com/lukas1001/firebase_auth_share)are my tests.
I hope this helps, if you have any questions feel free to ask ;)

**Reply by tony8864** [reply]
Thank you for your response! I just found this solution also..! Good luck!
